From 512702d6eb4d11ad74431e6fd61e25a7f642d498 Mon Sep 17 00:00:00 2001
From: Arien Judge <arien.judge@morsemicro.com>
Date: Mon, 4 Aug 2025 01:00:56 +1000
Subject: [PATCH 1/2] radsecproxy: update to 1.11.2

Refresh patches:
- 200-logdest-on-foreground.patch
- 300-uninit.patch

Changes:
  New features:
  - Native dynamic discovery for NAPTR and SRV records (#2, #83)
  - Optionally log accounting requests when respoinding directly (#72)
  - SNI support for outgoing connections (#90)
  - Optionally specify server name for certificate name check (#106)
  - Manual MTU setting for DTLS on non-linux platforms
  - TLS-PSK (#112)
  - Long hex-strings in config
  - Reload complete TLS context on SIGHUP, reload client/server cert and key (#76)
  - Implement SSLKEYLOGFILE mechanism
  - Options to require Message-Authenticator

  Misc:
  - Don't require server type to be set by dyndisc scripts
  - Improve locating openssl lib using pkg-config
  - Add Message-Authenticator to requests if missing
  - Re-verify certificates on SIGHUP and terminate invalid connections (#5)
  - Implement recommendations for deprecating insecure transports (#114)
  - verify EAP message content length (#128)
  - Close connection on radius attribute decode errors
  - Improve message-authenticator logging

  Bug Fixes:
  - Fix radius message length handling
  - Fix tcp connection not closed after idle timeout
  - Fix tls/dtls logging (#127, #132)
  - Fix dynamic connection not re-established after timeout if both auth- and accounting server
    are dynamically resolved (#131).
  - Fix error in dyndisc script result might cause radsecproxy to exit.
  - Fix some TLS config errors not reported at startup
  - Fix referencing non-existant rewrite passed without error
  - Fix correct secret for DTLS (radius/dtls)
  - Fix infinite loop when listening on tcp socket fails
  - Fix crashes under high load (#75, #109, #129)
  - Fix wrong DN in certificate request (#154)
  - Fix memory leak when using SIGHUP
  - Fix exit when dyndisc script returns illegal PSKkey
  - Fix logging during config check
  - Fix invalid realm configs are ignored
  - Fix default tls block selection
  - Fix Message-Authenticator validation for Accounting-Response

Signed-off-by: Arien Judge <arien.judge@morsemicro.com>
---
 net/radsecproxy/Makefile                        |  4 ++--
 .../patches/200-logdest-on-foreground.patch     | 17 +++++++++--------
 net/radsecproxy/patches/300-uninit.patch        |  2 +-
 3 files changed, 12 insertions(+), 11 deletions(-)

diff --git a/net/radsecproxy/Makefile b/net/radsecproxy/Makefile
index eec4cebdf..70f4179e8 100644
--- a/net/radsecproxy/Makefile
+++ b/net/radsecproxy/Makefile
@@ -8,12 +8,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=radsecproxy
-PKG_VERSION:=1.9.1
+PKG_VERSION:=1.11.2
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://github.com/radsecproxy/radsecproxy/releases/download/$(PKG_VERSION)/
-PKG_HASH:=e08e4e04d188deafd0b55b2f66b1e7fff9bdb553fb170846590317d02c9dc5db
+PKG_HASH:=1fe3f25a392b74db1fe62868e19e883acd1dc0e1f318715299920fcc5e166f97
 
 PKG_MAINTAINER:=Toke Høiland-Jørgensen <toke@toke.dk>
 PKG_LICENSE:=BSD-3-CLAUSE
diff --git a/net/radsecproxy/patches/200-logdest-on-foreground.patch b/net/radsecproxy/patches/200-logdest-on-foreground.patch
index 54c18f35a..c1aba626d 100644
--- a/net/radsecproxy/patches/200-logdest-on-foreground.patch
+++ b/net/radsecproxy/patches/200-logdest-on-foreground.patch
@@ -1,18 +1,19 @@
 --- a/radsecproxy.c
 +++ b/radsecproxy.c
-@@ -3075,15 +3075,13 @@ int radsecproxy_main(int argc, char **ar
- 	options.loglevel = loglevel;
+@@ -3637,16 +3637,13 @@ int radsecproxy_main(int argc, char **ar
+         options.loglevel = loglevel;
      else if (options.loglevel)
- 	debug_set_level(options.loglevel);
+         debug_set_level(options.loglevel);
 -    if (!foreground) {
--	debug_set_destination(options.logdestination
--                              ? options.logdestination
--                              : "x-syslog:///", LOG_TYPE_DEBUG);
--    	if (options.ftickssyslogfacility) {
+-        debug_set_destination(options.logdestination
+-                                  ? options.logdestination
+-                                  : "x-syslog:///",
+-                              LOG_TYPE_DEBUG);
+-        if (options.ftickssyslogfacility) {
 -            debug_set_destination(options.ftickssyslogfacility,
 -                                  LOG_TYPE_FTICKS);
 -            free(options.ftickssyslogfacility);
--    	}
+-        }
 +    debug_set_destination(options.logdestination
 +                          ? options.logdestination
 +                          : "x-syslog:///", LOG_TYPE_DEBUG);
diff --git a/net/radsecproxy/patches/300-uninit.patch b/net/radsecproxy/patches/300-uninit.patch
index 578c86826..5aeb0ac94 100644
--- a/net/radsecproxy/patches/300-uninit.patch
+++ b/net/radsecproxy/patches/300-uninit.patch
@@ -1,6 +1,6 @@
 --- a/gconfig.c
 +++ b/gconfig.c
-@@ -119,7 +119,7 @@ FILE *pushgconfpaths(struct gconffile **
+@@ -120,7 +120,7 @@ FILE *pushgconfpaths(struct gconffile **
      int i;
      FILE *f = NULL;
      glob_t globbuf;
-- 
2.25.1

